### ISSUE TITLE
Import of Cryptol tuples and records is broken

### ISSUE THREAD

**Issue by brianhuffman** [GaloisInc/saw-script#313]
When Cryptol terms involving operations on tuple and record types are imported using the cryptol quotes, it produces ill-typed terms that crash saw when evaluated. For example:
```
sawscript> check_term {{ (True, False) < (False, True) }}

user error ("check_term" (<stdin>:1:1-1:11):
Inferred type
  #{cmp :
  #(Prelude.Bool | #())
  -> #(Prelude.Bool | #())
  -> Prelude.Bool
  -> Prelude.Bool}
Not a subtype of expected type
  #{cmp :
  #(Prelude.Bool)
  -> #(Prelude.Bool)
  -> Prelude.Bool
  -> Prelude.Bool}
For term
  Cryptol.PCmpPair Prelude.Bool #() Cryptol.PCmpBit
  Cryptol.PCmpUnit
)
sawscript> print {{ (True, False) < (False, True) }}
saw: valPairLeft: Not a pair value: {1 = _, ...}
```
SAW quits to the command line after the `print` command. The result is similar when using record types, arithmetic operators, etc.

**Comment by brianhuffman** [comment]
Bisection shows that the first bad revision (that I could get to compile) is 83eeb932c22e1835b8d76c302632c380f9d83663. This and the preceding few commits brought in some updates to `cryptol-verifier` that are pretty obviously broken: For example, GaloisInc/cryptol-verifier@6fa9ab523245001699099de2e051fc6123893738 comments out a bunch of declarations from `Cryptol.sawcore` that are still explicitly mentioned by name in the translation in `Cryptol.hs`. The commented-out declarations are those having to do with the `Logic`, `Arith`, and `Cmp` dictionaries for record types.

**Comment by atomb** [comment]
Commit 9a26fb29 fixes class operations for tuples. I'll fix class operations for records in a later commit.

**Comment by brianhuffman** [comment]
This is now fixed in 8cbaa5adfba01e347e0e09692f2041fb97eeb552, and GaloisInc/cryptol-verifier@606c2f97e45abe5da02aa6d678a59aac64f92031. As of this version, Cryptol record types are  translated to plain tuple types in saw-core, i.e. the record field names are thrown away.
